File: insert_cll.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int data;
    struct Node* next;
} Node;

Node* insertIntoCLL(Node* head, int data) {
    Node* newn = (Node*)malloc(sizeof(Node));
    newn->data = data;
    if (!head) {
        newn->next = newn;
        return newn;
    }
    Node* cur = head;
    while (cur->next != head) cur = cur->next;
    cur->next = newn;
    newn->next = head;
    return head;
}

void printCLL(Node* head) {
    if (!head) { printf("Empty\n"); return; }
    Node* cur = head;
    do {
        printf("%d ", cur->data);
        cur = cur->next;
    } while (cur != head);
    printf("\n");
}

int main() {
    Node* head = NULL;
    head = insertIntoCLL(head, 1);
    head = insertIntoCLL(head, 2);
    head = insertIntoCLL(head, 3);
    printf("Circular Linked List: ");
    printCLL(head);
    return 0;
}

File: insert_cll.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def insert_into_cll(head, data):
    newn = Node(data)
    if not head:
        newn.next = newn
        return newn
    cur = head
    while cur.next != head:
        cur = cur.next
    cur.next = newn
    newn.next = head
    return head

def print_cll(head):
    if not head:
        print("Empty")
        return
    cur = head
    while True:
        print(cur.data, end=" ")
        cur = cur.next
        if cur == head:
            break
    print()

if __name__ == "__main__":
    head = None
    head = insert_into_cll(head, 1)
    head = insert_into_cll(head, 2)
    head = insert_into_cll(head, 3)
    print("Circular Linked List:", end=" ")
    print_cll(head)
